controllers/oneagent/daemonset.go

```diff
@@ -19,7 +19,7 @@ func prepareArgs(instance *dynatracev1alpha1.DynaKube, fs *dynatracev1alpha1.Ful
 	}
 
 	if feature == InframonFeature {
-		args = append(args, "--set-host-id-source=k8s-node-name")
+		args = append(args, "--set-host-id-source=auto")
```

Comment:
The other way around, when feature is not InframonFeature, then you set it to `auto`. When it's in infrastructure monitoring, it must be `k8s-node-name`